packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -50,7 +55,8 @@ BuildApp() {
   esac
 
   local framework_path="${FLUTTER_ROOT}/bin/cache/artifacts/engine/${artifact_variant}"
-  if [[ -n "$FLUTTER_FRAMEWORK_DIR" ]]; then
+  if [[ -n "$FLUTTER_FRAMEWORK_DIR" && "$ACTION" != "install" ]]; then
```

Comment:
Is this to prevent local engine overrides and then archive?

Reply:
It's because Xcode builds always pick up the previous build mode. 

Currently, it's necessary to run `flutter build ios --release` in order to produce a `Generated.xcconfig` with the correct build mode and framework directory for an archive build.